CONTRIBUTING.md

```diff
@@ -29,30 +31,138 @@ and maintained by Google](https://opensource.google.com/docs/cla/#why), not to
 ascribe any specific rights to a single OpenConfig contributor.
 
 ## Make a contribution
+
 To make a contribution to OpenConfig featureprofiles:
 
 1. Open a pull request in the
- [openconfig/featureprofiles](https://github.com/openconfig/featureprofiles) 
- repo. A brief description of the proposed addition along with references to 
+ [openconfig/featureprofiles](https://github.com/openconfig/featureprofiles)
+ repo. A brief description of the proposed addition along with references to
  any discussion issues should be included.
-    * Pull requests should be kept small. An ideal change is less than 500 lines. 
+    * Pull requests should be kept small. An ideal change is less than 500 lines.
      Small changes allow detailed discussions of the additions that are
      being made to the model, whilst also ensuring that course-corrections can be
      made early in the process. If a test is growing to more than 500 lines, it
      may need to be broken into multiple smaller tests.
 
-1. The pull request should include both the suggested feature profile textproto 
+1. The pull request should include both the suggested feature profile textproto
  additions, as well as any relevant additions to tests. Tests should be written
  in golang using the [ONDATRA](https://github.com/openconfig/ondatra) framework.
 
 1. The automated CI running against each pull request will check the pull
  request for compliance.  The author should resolve any issues found by CI.
 
-1. One or more peers in the community may review the pull request.   
+1. One or more peers in the community may review the pull request.
 
 1. A feature profile repository maintainer will be reponsible for a final review
-and approval.  Only a feature repository maintainer can merge a pull request to 
+and approval.  Only a feature repository maintainer can merge a pull request to
 the main branch.
   
-The aim of this process is maintain the model quality and approach that OpenConfig 
+The aim of this process is maintain the model quality and approach that OpenConfig
 working group has strived for since its inception in 2014. Thank you for your contributions!
+
+## Test development style guide
+
+### Directory Organization
+
+The directory tree is organized as follows:
+
+* `cloudbuild/` contains google cloud build scripts for running virtual
+    routers in containers on [KNE](https://github.com/google/kne)
+* `feature/` contains definition and tests of feature profiles.
+* `internal/` contains packages used by feature profile tests.
+* `proto/`  contains protobuf files for feature profiles.
+* `tools/` contains code used for CI checks.
+* `topologies/` contains the testbed topology definitions.
+* `yang/` contains autogenerated go structures.
+
+### Test Suite Organization
+
+Under `feature/<featurename>/[<sub-feature>/]tests`, each feature group has its own
```

Comment:
If there are multiple tests for a given subfeature, we should put them into separate subdirectories, otherwise they would belong to the same package and would execute together. Each test can already take 30+ minutes.

So for example, it should be:

* `feature/interfaces/tests/rt_5_1_singleton_test/main_test.go`
* `feature/interfaces/tests/rt_5_2_aggregate_test/main_test.go`

(Test files need to end with `_test.go` otherwise `go test` would not recognize them.)